diagnostic_feedback/validators/base_validator.py

```diff
@@ -0,0 +1,53 @@
+
+
+class BaseValidator(object):
+    """
+    Generic method' for validation
+    """
+    xblock = None
+    _ = None
+
+    def __init__(self, xblock):
+        self.xblock = xblock
+        self._ = xblock._
+
+    def is_empty(self, value):
+        """
+        check for empty string
+        :param value: string to test
+        :return: Boolean
+        """
+        return not value.strip()
+
+    def invalid_url(self, url):
+        """
+        check for valid url
+        :param value: url to test
+        :return: Boolean
+        """
+        return url.strip() and not url.strip().startswith('http')
```

Comment:
If `url` only contains whitespace, this will return an empty string instead of a boolean.  And since an empty string is falsy in python, it will report the (empty) URL as valid.